### PR TITLE
Added Apple Silicon compatibility with onnxruntime

### LINKED ISSUE #15: onnxruntime_gpu unavailable for Apple Silicon
**Problem:** Cannot install `onnxruntime_gpu==1.15.1` on Apple silicone (M2)

```bash
> pip install -r requirements.txt
Collecting numpy==1.23.5 (from -r requirements.txt (line 1))
  Using cached numpy-1.23.5-cp310-cp310-macosx_11_0_arm64.whl (13.4 MB)
ERROR: Could not find a version that satisfies the requirement onnxruntime_gpu==1.15.1 (from versions: none)
ERROR: No matching distribution found for onnxruntime_gpu==1.15.1
```

**Solution:**
Use `onnxruntime-silicon` instead.

**Local environment:**
```
System Software Overview:

  System Version: macOS 14.1.2 (23B92)
  Kernel Version: Darwin 23.1.0
  Boot Volume: Macintosh HD
  CPU: Apple M2 Pro
  Boot Mode: Normal
```

### PR DESCRIPTION
**Pull request by kevinlacaille**
Closes #15 

**TL;DR:** Replaced `requirements.txt` with `setup.py` and allow for Apple Silicon compatibility.

**Summary:**

I made an important update to the installation process to improve the compatibility of WALDO for Apple Silicon processors.

1. I removed the `requirements.txt` file and replaced it with a `setup.py` file, which allows for multiple options for a given dependency. In this case, to ensure compatibility with Apple Silicon, I've replaced the previous `onnxruntime-gpu` dependency with `onnxruntime-silicon`, which can be found on PyPI [here](https://pypi.org/project/onnxruntime-silicon/). 
    - Please feel free to change or ask for revisions to any of the parameters in the `setup` function (i.e., description, classifiers, etc.)
3. I updated the README to reflect these changes and provide clear instructions on how to install the project.

### CODE REVIEW

**Comment by stephansturges** [comment]
Awesome, thanks!
